### PR TITLE
work on fixing missing strings and addressing TM failings

### LINKED ISSUE #5282: Strings in single group sub-tab not displaying localized
### Observed behavior

Some strings on group management page and modals do not display localized, despite being translated and approved on Crowdin:

[String translated on Crowdin](https://crowdin.com/translate/kolibri/582/en-es#43232)

### Expected behavior
Localized strings displayed correctly.

### User-facing consequences
Lack of understanding the UI options.

### Errors and logs
…

### Steps to reproduce
Change UI language from English
Go to Coach > Plan > Group and select a group. 
Try to enroll and delete learners.

### Context
0.12.1 (kolibridemo server)

cc @indirectlylit @jonboiser

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

* migrate a bunch of strings that hadn't been picked up to use generic shared strings. Why weren't they picked up? looks like a combination of
   * cross-component-cross-plugin issues
   * at least one string added after freeze
   * issues with defining translators in mixins
* work on tweaking upload and translation memory processes to maintain currently-approved strings

the new strings are in crowdin under `release-v0.12.2`:
https://crowdin.com/project/kolibri/settings#files

### Reviewer guidance

I have given this a cursory test but it could use more evaluation

### References

fixes #5282

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

File: packages/kolibri-tools/package.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
     "webpack-dev-server": "^3.1.14",
     "webpack-hot-middleware": "2.12.1",
     "webpack-merge": "^4.1.0",
-    "webpack-rtl-plugin": "^1.7.0"
+    "webpack-rtl-plugin": "^2.0.0"
```
Comment (rtibbles) [inline packages/kolibri-tools/package.json:85]: Woot!

**Review by rtibbles — COMMENTED** [review]
Code changes make sense to me - manual testing needed, ofc!

**Comment by radinamatic** [comment]
Ok, just restarted the build on Buildkite, and this time it was successful!
Will give it a whirl 🙂

**Comment by radinamatic** [comment]
Not able to change languages (Windows 10):

### Setup wizard


### Upon installation

**Comment by radinamatic** [comment]
Trying to tun the `kolibri language setdefault` command, not sure if that is related to the errors above:

```
C:\Python34\Scripts>kolibri language setdefault ar
[32mINFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE[0m
[37mINFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base[0m
```

Apparently nothing more happened, so I try Spanish:

```
C:\Python34\Scripts>kolibri language setdefault es
[32mINFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE[0m
Traceback (most recent call last):
  File "C:\Python34\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python34\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python34\Scripts\kolibri.exe\__main__.py", line 9, in <module>
  File "C:\Python34\lib\site-packages\kolibri\utils\cli.py", line 630, in main
    initialize(debug=debug)
  File "C:\Python34\lib\site-packages\kolibri\utils\cli.py", line 170, in initialize
    django.setup()
  File "C:\Python34\lib\site-packages\kolibri\dist\django\__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\apps\registry.py", line 108, in populate
    app_config.import_models()
  File "C:\Python34\lib\site-packages\kolibri\dist\django\apps\config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "C:\Python34\lib\importlib\__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "C:\Python34\lib\site-packages\kolibri\dist\django\contrib\auth\models.py", line 103, in <module>
    class Group(models.Model):
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\models\base.py", line 162, in __new__
    new_class.add_to_class(obj_name, obj)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\models\base.py", line 325, in add_to_class
    value.contribute_to_class(cls, name)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\models\fields\related.py", line 1648, in contribute_to_class
    self.remote_field.through = create_many_to_many_intermediary_model(self, cls)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\models\fields\related.py", line 1104, in create_many_to_many_intermediary_model
    'verbose_name': _('%(from)s-%(to)s relationship') % {'from': from_, 'to': to},
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\functional.py", line 162, in __mod__
    return six.text_type(self) % rhs
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\functional.py", line 119, in __text_cast
    return func(*self.__args, **self.__kw)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\__init__.py", line 89, in ugettext
    return _trans.ugettext(message)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 338, in gettext
    return do_translate(message, 'gettext')
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 321, in do_translate
    _default = _default or translation(settings.LANGUAGE_CODE)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 228, in translation
    _translations[language] = DjangoTranslation(language)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 131, in __init__
    self._add_local_translations()
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 183, in _add_local_translations
    translation = self._new_gnu_trans(localedir)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\utils\translation\trans_real.py", line 156, in _new_gnu_trans
    fallback=use_null_fallback)
  File "C:\Python34\lib\gettext.py", line 410, in translation
    t = _translations.setdefault(key, class_(fp))
  File "C:\Python34\lib\gettext.py", line 160, in __init__
    self._parse(fp)
  File "C:\Python34\lib\gettext.py", line 300, in _parse
    catalog[str(msg, charset)] = str(tmsg, charset)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd8 in position 0: ordinal not in range(128)

C:\Python34\Scripts>
```

Now running even `kolibri start` gives the above error, and server is not starting with the desktop shortcut either... 😕

**Comment by radinamatic** [comment]
Same issue on Ubuntu 16.04:

### Setup wizard


### Upon installation

**Comment by radinamatic** [comment]
Stopped the service and run from Terminal:




```
INFO:cherrypy.access.140715085888200:127.0.0.1 - - [21/Mar/2019:23:29:04] "GET /static/default_frontend/kolibri-logo.svg?625360a31b0628f83d4ca2ef861dbd30 HTTP/1.1" 200 10220 "http://127.0.0.1:8080/device/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0"
INFO:cherrypy.access.140715085888200:127.0.0.1 - - [21/Mar/2019:23:29:05] "GET /static/default_frontend/default_frontend-0.12.2a0.dev0+git.27.g3d33970e.js.map HTTP/1.1" 304 - "" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0"
ERROR    Internal Server Error: /i18n/setlang/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/core/decorators.py", line 275, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/core/views.py", line 46, in set_language
    next_trans = translate_url(next, lang_code)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/base.py", line 177, in translate_url
    with override(lang_code):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/__init__.py", line 181, in __enter__
    activate(self.language)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/__init__.py", line 166, in activate
    return _trans.activate(language)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 239, in activate
    _active.value = translation(language)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 228, in translation
    _translations[language] = DjangoTranslation(language)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 131, in __init__
    self._add_local_translations()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 183, in _add_local_translations
    translation = self._new_gnu_trans(localedir)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 156, in _new_gnu_trans
    fallback=use_null_fallback)
  File "/usr/lib/python3.5/gettext.py", line 441, in translation
    t = _translations.setdefault(key, class_(fp))
  File "/usr/lib/python3.5/gettext.py", line 162, in __init__
    self._parse(fp)
  File "/usr/lib/python3.5/gettext.py", line 316, in _parse
    catalog[str(msg, charset)] = str(tmsg, charset)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd8 in position 0: ordinal not in range(128)

```

**Comment by rtibbles** [comment]
Ah, our old friend UnicodeDecodeError. Guessing this might be because we are missing some unicode handling in a file.

**Comment by rtibbles** [comment]
Weird that this is happening on Python3 though.

**Comment by radinamatic** [comment]
Seems so...

```
osboxes@osboxes:~/Desktop$ kolibri language setdefault es
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
WARNING:root:Invalid language code es. Must be one of: ['ar', 'bg-bg', 'bn-bd', 'en', 'es-es', 'fa', 'fr-fr', 'ff-cm', 'hi-in', 'mr', 'my', 'nyn', 'pt-br', 'sw-tz', 'te', 'ur-pk', 'vi', 'yo']
osboxes@osboxes:~/Desktop$ kolibri language setdefault ar
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
osboxes@osboxes:~/Desktop$ kolibri start --debug --foreground
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
Traceback (most recent call last):
  File "/usr/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri==0.12.2a0.dev0+git.27.g3d33970e', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 630, in main
    initialize(debug=debug)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 170, in initialize
    django.setup()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/apps/registry.py", line 108, in populate
    app_config.import_models()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/contrib/auth/models.py", line 103, in <module>
    class Group(models.Model):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 162, in __new__
    new_class.add_to_class(obj_name, obj)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 325, in add_to_class
    value.contribute_to_class(cls, name)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/fields/related.py", line 1648, in contribute_to_class
    self.remote_field.through = create_many_to_many_intermediary_model(self, cls)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/fields/related.py", line 1104, in create_many_to_many_intermediary_model
    'verbose_name': _('%(from)s-%(to)s relationship') % {'from': from_, 'to': to},
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/functional.py", line 162, in __mod__
    return six.text_type(self) % rhs
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/functional.py", line 119, in __text_cast
    return func(*self.__args, **self.__kw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/__init__.py", line 89, in ugettext
    return _trans.ugettext(message)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 338, in gettext
    return do_translate(message, 'gettext')
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 321, in do_translate
    _default = _default or translation(settings.LANGUAGE_CODE)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 228, in translation
    _translations[language] = DjangoTranslation(language)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 131, in __init__
    self._add_local_translations()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 183, in _add_local_translations
    translation = self._new_gnu_trans(localedir)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/translation/trans_real.py", line 156, in _new_gnu_trans
    fallback=use_null_fallback)
  File "/usr/lib/python3.5/gettext.py", line 441, in translation
    t = _translations.setdefault(key, class_(fp))
  File "/usr/lib/python3.5/gettext.py", line 162, in __init__
    self._parse(fp)
  File "/usr/lib/python3.5/gettext.py", line 316, in _parse
    catalog[str(msg, charset)] = str(tmsg, charset)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd8 in position 0: ordinal not in range(128)
osboxes@osboxes:~/Desktop$ 
```

**Comment by indirectlylit** [comment]
very sorry guys - I had unpushed changes

**Comment by radinamatic** [comment]
@indirectlylit Strings in *Groups* subtab look good on Ubuntu, still some icons overlap names, but I guess that's a different fix:

**Comment by indirectlylit** [comment]
@radina tested, looks good now

**Pull request merged**